chromium_src/components/sync_bookmarks/bookmark_specifics_conversions.cc

```diff
@@ -16,10 +16,11 @@ namespace sync_bookmarks {
 sync_pb::EntitySpecifics CreateSpecificsFromBookmarkNode(
     const bookmarks::BookmarkNode* node,
     bookmarks::BookmarkModel* model,
-    bool force_favicon_load) {
+    bool force_favicon_load,
+    bool include_guid) {
   brave_sync::AddBraveMetaInfo(node);
-  return CreateSpecificsFromBookmarkNodeChromiumImpl(node, model,
-                                                     force_favicon_load);
+  return CreateSpecificsFromBookmarkNodeChromiumImpl(
+      node, model, force_favicon_load, include_guid);
```

Comment:
cc @darkdh

Reply:
signature changes is ok but we don't have a way to propagate it v1 so we leave it empty when convert brave record to bookmark specific. Before the [upstream change](https://chromium.googlesource.com/chromium/src/+/670c996988f9e37c2b0bc809338bd893e6355af), it was fine because it had a valid GUID check
We might still need this condition check `base::IsValidGUID(update_entity.specifics.bookmark().guid())` in 
```cpp
-  // If there is a different GUID in the specifics and it is valid, we must
-  // replace the entire node in order to use it, as GUIDs are immutable. Further
-  // updates are then applied to the new node instead.
-  if (update_entity.specifics.bookmark().guid() != node->guid() &&
-      base::IsValidGUID(update_entity.specifics.bookmark().guid())) {
-    const bookmarks::BookmarkNode* old_node = node;
-    node = ReplaceBookmarkNodeGUID(
-        node, update_entity.specifics.bookmark().guid(), model);
-    tracker->UpdateBookmarkNodePointer(old_node, node);
+  // The GUID is immutable and cannot change. This check is somewhat redundant
+  // here because there is a similar one also accounted to kUnexpectedGuid, but
+  // in theory a misbehaving server could trigger this codepath.
+  // TODO(crbug.com/1032052): If the tracker becomes client-tag-centric, this
+  // shouldn't be necessary.
+  if (update_entity.specifics.bookmark().guid() != node->guid()) {
+    LogProblematicBookmark(RemoteBookmarkUpdateError::kUnexpectedGuid);
+    return;
   }
+
``` 
and 
```cpp
+
+    // This may be a good chance to populate the client tag for the first time.
+    // TODO(crbug.com/1032052): Remove this code once all local sync metadata
+    // is required to populate the client tag (and be considered invalid
+    // otherwise).
+    if (tracked_entity && !tracked_entity->has_final_guid() &&
+        !update_entity.is_deleted() &&
+        update_entity.server_defined_unique_tag.empty()) {
+      DCHECK(base::IsValidGUID(update_entity.specifics.bookmark().guid()));
+      bookmark_tracker_->PopulateFinalGuid(
+          update_entity.id, update_entity.specifics.bookmark().guid());
+    }
+
```
otherwise, new updates will not be applied. cc @AlexeyBarabash

Reply:
@darkdh : Are these changes in the `master` or something being worked on? I don't see them in the current `components-sync_bookmarks-bookmark_remote_updates_handler.cc.patch`.

Reply:
no, we don't need patch in current master now but if we upgrade to `81.0.4044.62`, we will need the patch

Reply:
@darkdh it looks like [this upstream change](https://chromium.googlesource.com/chromium/src/+/0866160e551d5958a94e4d03feaaf965740f432b%5E%21/#F2) reverted the first patch and more changes to the second.

Reply:
@mkarolin yes, and that is why it will break our sync v1 because we don't have valid guid in bookmarks specifics

Reply:
Ok, seems like we are good for now. DM'd with @darkdh and .69 has what we need. Will have to watch out for the consequent bumps as more changes have been committed for that file, not sure what will end up in cr81.